portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructureee/search/searchexperiences/Blueprints.testcase

```diff
@@ -1157,6 +1157,128 @@ definition {
 		AssertElementPresent(locator1 = "Blueprints#BLUEPRINTS_ADDED_ELEMENTS_EMPTY");
 	}
 
+	@description = "This is a use case for LPS-185340 - Configure Collection Display that returns Blueprint assets."
+	@priority = 3
+	test ConfigureCollectionDisplayThatReturnsBlueprintAssets {
+		property custom.properties = "feature.flag.LPS-129412=true";
+
+		task ("Create the Pages Blueprint with the Page query setting type") {
+			Blueprints.openBlueprintsAdmin();
+
+			Blueprints.createBlueprint(title = "Pages");
+
+			Navigator.gotoNavNested(navNested = "Query Settings");
+
+			Check.checkNotVisible(
+				locator1 = "Radio#ANY",
+				radioLabel = "Selected Types");
+
+			Button.click(button = "Select Asset Types");
+
+			Check.checkNotVisible(
+				key_assetType = "Page",
+				locator1 = "Blueprints#BLUEPRINTS_SEARCHABLE_ASSET_TYPE_CHECKBOX");
+
+			Button.clickDone();
+
+			PortletEntry.save();
+		}
+
+		task ("Create the Web Content Articles Blueprint with the Web Content Article query setting type and create a few web contents") {
+			Blueprints.openBlueprintsAdmin();
+
+			Blueprints.createBlueprint(title = "Web Content Articles");
+
+			Navigator.gotoNavNested(navNested = "Query Settings");
+
+			Check.checkNotVisible(
+				locator1 = "Radio#ANY",
+				radioLabel = "Selected Types");
+
+			Button.click(button = "Select Asset Types");
+
+			Check.checkNotVisible(
+				key_assetType = "Web Content Article",
+				locator1 = "Blueprints#BLUEPRINTS_SEARCHABLE_ASSET_TYPE_CHECKBOX");
+
+			Button.clickDone();
+
+			PortletEntry.save();
+
+			JSONWebcontent.addWebContent(
+				content = "Some content about Apples",
+				groupName = "Guest",
+				title = "Apples");
+		}
+
+		task ("Create two Pages and configure with Page and Web Content Article collection displays") {
+			for (var count : list "1,2") {
+				JSONLayout.addPublicLayout(
+					groupName = "Guest",
+					layoutName = "Test Page ${count}",
+					type = "content");
+
+				JSONLayout.publishLayout(
+					groupName = "Guest",
+					layoutName = "Test Page ${count}");
+			}
```

Comment:
Let's make these page names more descriptive. Maybe "Page Collection Display Page" and "Web Content Collection Display Page"?